.github/CONTRIBUTING.md

```diff
@@ -17,40 +16,103 @@ We prefer questions and support requests be posted in the [forum](http://forum.t
 
 __Please provide as much information as possible to help us solve problems and answer questions better!__
 
-Commits and Commit Messages
----------------------------
-Follow these guidelines when creating public commits and writing commit messages.
+## PR Title / Commit Message Guidelines
+
+We try to follow the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0-beta.2/) style for commit messages and pull request titles.
+This leads to **more readable messages** that are easy to follow when looking through the **project history**.
+But also, we use the git commit messages to **generate the Terasology change log**.
+
+### Commit Message Format
+Each commit message consists of a **header**, a **body** and a **footer**.  The header has a special
+format that includes a **type**, a **scope** and a **subject**:
+
+```
+<type>(<scope>): <subject>
+<BLANK LINE>
+<body>
+<BLANK LINE>
+<footer>
+```
+
+The **header** is mandatory and the **scope** of the header is optional.
+
+Any line of the commit message cannot be longer 100 characters! This allows the message to be easier
+to read on GitHub as well as in various git tools.
+
+The footer should contain a [closing reference to an issue](https://help.github.com/articles/closing-issues-via-commit-messages/) if any.
+
+Samples: (even more [samples](https://github.com/angular/angular/commits/master))
+
+```
+docs(changelog): update changelog to beta.5
+```
+```
+fix(release): need to depend on latest rxjs and zone.js
+
+The version in our package.json gets copied to the one we publish, and users need the latest of these.
+```
+
+### Revert
+If the commit reverts a previous commit, it should begin with `revert: `, followed by the header of the reverted commit. In the body it should say: `This reverts commit <hash>.`, where the hash is the SHA of the commit being reverted.
+
+### Type
+Must be one of the following:
+
+* **build**: Changes that affect the build system or external dependencies (example scopes: gradle, git, idea)
+* **ci**: Changes to our CI configuration files and scripts (example scopes: Jenkins)
```

Comment:
Would suggest "logistics" instead. That way it can count any sort of meta-work that's about the project and its tooling. CI is somewhat narrow in meaning, IMHO, specifically being for instance the `Jenkinsfile` - changes to Gradle aren't necessarily specific to CI, for instance. Utility scripting, template files, `.gitignore` are other examples.

Reply:
Gradle changes would fall into the `build` category above... would you rather merge those two into a single entry `logistics`?

```
logistics(gradle): ... 
logistics(git): ...
logistics(idea): ...
logistics(jenkins): ...
```

Reply:
Reviewing this list to pick the right type and scope, I'm noticing that I can be doing "build" (gradle) things that _also_ could be categorized as doc/fix/perf/refactor/style/test.

Which makes gradle feel more like a scope than a type?

On the other hand, it seems like a good idea to be able to separate build stuff from things that would be relevant to people who only use the distribution and don't do source builds. (doesn't really apply to engine as it is today, but in principle.)

Reply:
I agree, and I think it is up to us to define what we want to achieve by these guidelines (for PR titles). The _types_ and _scopes_ are a bit mixed up (e.g., you could argue about `doc: fix typo` vs `fix: correct typo in doc`, although this might exaggerating).

This is my personal view on the these _types_ in a bit more detail:
- **build** and **ci** affect the build process either for local development or our automation in the "cloud". This may have an impact on the final game (as seen by players), but is mostyl under the hood. Obviosuly, you could do `perf` or `fix` changes for these _types_ ...
- **feat**, **fix**, **perf** are noticable by players and thus have a direct impact on our final product. They may be more or less noticable in practice.
- **test** indicates that we worked on tests - which is nothing visible to players. This can be tricky, as updating/fixing a test often also requires to fix behavior in the code. In this case, I'd go for a `test: ...` commit message, but use **fix: ...** as PR title.
- **doc** if we only udpate Javadoc, or update documents living in the repository (e.g., in NUI)
- **refactor** and **style** should not change the behavior of the code and should not have an impact on anything... We currently use `chore` in a similar fashion, maybe that would fit better? What do you think?

@keturn @Cervator @jdrueckert what do you think?

Reply:
I might be too distant from this to really have a strong opinion I'd want people to listen to - so please don't take this too seriously. I thought the base conventional commits bit was interesting, going particularly deep or even using the (something) approach to indicate finer details is beyond the point that really concerned me. Not even truly sure the first thing (feat fix etc) is _super_ meaningful in my head, it was just novel to have _some_ sort of convention, I guess? Probably why I'm reluctant to wrap a whole lot of rigidity around it all, with a more laissez faire style fitting my level of interest.

That's just me though - I'm not saying that can't be useful, or even hugely impactful particularly if you fully control the environment and can count on everybody using the convention perfectly to the point where you can really wrap some cool automation around it. In our case I just worry the overhead from the extra process won't be worth the benefit. So in the end it is nice if there's a little more consistency to commits, but I spend so very little time looking at them anyway ... 🤷